en/python_installation/instructions.md

```diff
@@ -16,13 +16,13 @@ First check whether your computer is running a 32-bit version or a 64-bit versio
 
 You can download Python for Windows from the website https://www.python.org/downloads/windows/. Click on the "Latest Python 3 Release - Python x.x.x" link. If your computer is running a **64-bit** version of Windows, download the **Windows x86-64 executable installer**. Otherwise, download the **Windows x86 executable installer**. After downloading the installer, you should run it (double-click on it) and follow the instructions there.
 
-One thing to watch out for: During the installation, you will notice a window marked "Setup". Make sure you tick the "Add Python 3.6 to PATH" or 'Add Python to your environment variables" checkbox and click on "Install Now", as shown here (it may look a bit different if you are installing a different version):
+One thing to watch out for: During the installation, you will notice a window marked "Setup". Make sure you tick the "Add Python {{ book.py_version }} to PATH" or 'Add Python to your environment variables" checkbox and click on "Install Now", as shown here (it may look a bit different if you are installing a different version):
 
 ![Don't forget to add Python to the Path](../python_installation/images/python-installation-options.png)
 
 When the installation completes, you may see a dialog box with a link you can follow to learn more about Python or about the version you installed. Close or cancel that dialog -- you'll be learning more in this tutorial!
 
-Note: if you are using an older version of Windows (7, Vista, or any older version) and the Python 3.6.x installer fails with an error, you can try either:
+Note: if you are using an older version of Windows (7, Vista, or any older version) and the Python {{ book.py_version }}.x installer fails with an error, you can try either:
 1. install all Windows Updates and try to install Python again; or
 2. install an [older version of Python](https://www.python.org/downloads/windows/), e.g., [3.4.6](https://www.python.org/downloads/release/python-346/).
```

Comment:
[Django 3.1 requires Python 3.6 or greater](https://docs.djangoproject.com/en/3.1/faq/install/#faq-python-version-support), so we shouldn't recommend Python 3.4.6 anymore, nor use it as an example.

Does the issue of installers for newer Python versions failing for older Windows versions still occur at all? If so, do we still have to assume some participants are using these Windows versions? (Do they still get security updates?) And does trying with an older version of Python have any chance of success, if that version would have to be 3.6 or greater, and the previous version of this instruction assumed the issue would occur with the Python 3.6 installer?

Reply:
@das-g Found current compatibility versions for Windows

[Python 3.4.10](https://www.python.org/downloads/release/python-3410/) is the last python version that can be used in Windows XP or earlier as per Python.org. It is also reached its end of development. 

[Python 3.8.6](https://www.python.org/downloads/release/python-386/) is the last python version that can be used in Windows 7 or earlier as per Python.org.

If we can assume that no one is using Windows XP or earlier to learn Django using Djangogirls, then we can recommend, Python 3.8.6 as the latest version (Python 3.9.0 which came afterwards have dropped support for Window 7 or earlier versions)

So using version 3.8.6 is a better choice for Django girls in general. But Pythonanywhere is using Python 3.8.0. I don't think there will be much issues here with different patch versions. Version 3.8 in general will be safe choice I think.

In that case, we don't have to worry about backward compatibility of Python with Windows 7 or Vista.

Reply:
@das-g Check commit https://github.com/DjangoGirls/tutorial/commit/d90b4a38a4531b28701c866cff611f1cc7c8161e

Can I have confirmation about the following change on python version?

```
{
    "gitbook": ">=3.2.0",
    "variables": {
      "py_version": "3.8", # Previously 3.9
      "py_release": "3.8.6", # Previously 3.9.0
      "py_min_version": "3.6.0",
      "pa_py_version": "3.8",
      "django_version": "2.2.4"
    },

...
```

Reply:
> Can I have confirmation about the following change on python version?
> 
> ```
>       ...
>       "py_version": "3.8", # Previously 3.9
>       "py_release": "3.8.6", # Previously 3.9.0
>       ...
> ```

:+1: fine with me

Reply:
> Does the issue of installers for newer Python versions failing for older Windows versions still occur at all? 

With version 3.8.6, it still supports  Windows 7 and Vista as per Python.org

> If so, do we still have to assume some participants are using these Windows versions? (Do they still get security updates?)

We have to assume that our participants are not using the Windows version released before 7 or Vista versions. Version 3.8 still has its security updates comming until October of 2024 as per [Python.org](https://www.python.org/dev/peps/pep-0569/#id4) and [DevGuide](https://devguide.python.org/#status-of-python-branches)

>And does trying with an older version of Python have any chance of success, if that version would have to be 3.6 or greater, and the previous version of this instruction assumed the issue would occur with the Python 3.6 installer?

From the information I gathered, I came to know that only version 3.4 have further backward compatibilty. Versions 3.6 and 3.8 should have same result as per Python.org. Since we can't support 3.4 with Django 3.1 in our hands, there is a last resort kind of solution. 

Create a copy of the current version of Django (2.2.4) and Publish a new subdomain like "windows.djangogirls.org". So that this url will act as an archive and help users who have older versions of Windows.

@das-g Check the above suggestion.

Reply:
> We have to assume that our participants are not using the Windows version released before 7 or Vista versions.

Yeah. My question is: Is that a suitable assumption?

> Version 3.8 still has its security updates comming until October of 2024 as per Python.org and DevGuide

Sorry, what I meant was: Do _these Windows versions_ still receive Windows security updates from Microsoft?

Any Windows version that doesn't receive security updates anymore can be considered unsupported by Microsoft, so we probably don't have to support it either. (As those still using it should update to a newer version or change to some other operating system anyway, lest their system is left in undue risk.)

> Since we can't support 3.4 with Django 3.1 in our hands, there is a last resort kind of solution.
> 
> Create a copy of the current version of Django (2.2.4) and Publish a new subdomain like "windows.djangogirls.org". So that this url will act as an archive and help users who have older versions of Windows.

I'd rather not go that way, as we struggle enough to keep one version of the tutorial maintained.

Reply:
> Yeah. My question is: Is that a suitable assumption?

If you are a person who is working on Windows XP, 100% of the things can go wrong badly while installing. Every in windows 7, new softwares are very hard to install without Windows updateds.

> Sorry, what I meant was: Do _these Windows versions_ still receive Windows security updates from Microsoft?

https://en.wikipedia.org/wiki/List_of_Microsoft_Windows_versions

As per the above link, Microsoft ended support for all versions before Windows 8.1 on the consumer side. Which means Windows 8, 7, Vista, XP, etc. So currently Microsoft is only providing updates for Windows 8.1 and Windows 10.

> Any Windows version that doesn't receive security updates anymore can be considered unsupported by Microsoft, so we probably don't have to support it either. (As those still using it should update to a newer version or change to some other operating system anyway, lest their system is left at undue risk.)

As per Python.org, 3.6.0 does not support any windows version of Windows XP or earlier. Django 3+ need Python 3.6.0. So we can reach a logical conclusion that users need to follow minimal requirement for using this tutorial as Windows 7 or later since we can support it even without Microsoft support. But If you want only support Windows versions with MS updates, then we have to stick to Windows 8.1 and Windows 10 as mentioned above.

Fun fact: https://stackoverflow.com/questions/53821523/how-to-deploy-a-django-app-locally-on-a-windows-xp-machine

Reply:
Btw, **Python 3.6.12** is the lastest under Python 3.6. Our current version set to 3.6.0. Should we change it to 3.6.12 which has all the new patchs?

Reply:
I just noticed that Python version 3.6.12 does not have any downloadable files for windows. Only version [3.6.8](https://www.python.org/downloads/release/python-368/) has all the downloadable files for Windows as well as macOS.

Reply:
Resolved with commit https://github.com/DjangoGirls/tutorial/pull/1683/commits/3eacfdef9efbb4d18b8db3c2762a7c9cf18e0088

Reply:
https://www.python.org/downloads/release/python-3612/ mentions
>  Python 3.6.8 was the last bugfix release for 3.6. **Python 3.6 is now in the security fix phase** of its life cycle. Only security-related issues are accepted and addressed during this phase. We plan to provide security fixes for Python 3.6 as needed through 2021, five years following its initial release. **Security fix releases** are produced periodically as needed and **are source-only releases; binary installers are not provided.**

(emphasis mine)

I guess security fix releases are meant mostly for old but still supported releases of Linux distributions that package an old minor version of Python, and therefore don't require upstream binaries. (As Linux distributions tend to compile their own binaries from the upstream source code.) The idea is probably that Windows users can simply get a later minor version instead, which seems true for all still-maintained versions of Windows.